core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -389,14 +389,17 @@ public void addSplits(String tableName, SortedSet<Text> partitionKeys) throws Ta
         if (exception.get() != null) {
           executor.shutdownNow();
           Throwable excep = exception.get();
-          if (excep instanceof TableNotFoundException)
-            throw (TableNotFoundException) excep;
-          else if (excep instanceof AccumuloException)
-            throw (AccumuloException) excep;
-          else if (excep instanceof AccumuloSecurityException)
-            throw (AccumuloSecurityException) excep;
+          if (excep instanceof TableNotFoundException) {
+            TableNotFoundException tnfe = (TableNotFoundException) excep;
+            throw new TableNotFoundException(tableId, tableName, "Table not found by background thread", tnfe);
+          } else if (excep instanceof AccumuloSecurityException) {
+            // base == background accumulo security exception
+            AccumuloSecurityException base = (AccumuloSecurityException) excep;
+            throw new AccumuloSecurityException(base.getUser(), base.asThriftException().getCode(), base.getTableInfo(), excep);
+          } else if (excep instanceof Error)
+            throw new Error(excep);
```

Comment:
Thats the case I was rambling about earlier.  Because if the user prints this stack trace, they will not see the code path that called TablerOps.addSplits().  They will just see the stack trace for some background thread that TablerOps.addSplits() created.  So if your code has multiple places that call  TablerOps.addSplits(), then you may not know which one triggered the exception.